actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -868,6 +868,7 @@ def recognize_path_with_request(req, path, extras, raise_on_missing: true)
           if app.matches?(req) && app.dispatcher?
             begin
               req.controller_class
+              req.action_name
```

Comment:
I understand why you've added it, but I don't see any reason this is needed / has any effect

Reply:
Thank you for reviewing this. I removed this method call.